source_http.go

```diff
@@ -43,13 +50,27 @@ func (s *HttpImageSource) fetchImage(url *url.URL) ([]byte, error) {
 		return nil, fmt.Errorf("Error downloading image: (status=%d) (url=%s)", res.StatusCode, req.URL.String())
 	}
 
+	// Read the body
 	buf, err := ioutil.ReadAll(res.Body)
 	if err != nil {
 		return nil, fmt.Errorf("Unable to create image from response body: %s (url=%s)", req.URL.String(), err)
 	}
 	return buf, nil
 }
 
+func (s *HttpImageSource) setAuthorizationHeader(req *http.Request, ireq *http.Request) {
+	auth := s.Config.Authorization
```

Comment:
In certain situations, it might be useful to override the Authorization Header command line value with the one found on X-Forward-Authorization. While the present solution removes this flexibility it also eliminates a brute force vector attack which uses imaginary as proxy for trying out different Authorization combinations. Yet again I would argue that imaginary should not be concern with this particular security issue as I would expect production usage to involve 3rd party firewall app.